src/shogun/kernel/CombinedKernel.cpp

```diff
@@ -169,15 +176,99 @@ bool CCombinedKernel::init(CFeatures* l, CFeatures* r)
 		return false;
 	}
 
-	if ( ((CCombinedFeatures*) l)->get_num_feature_obj()<=0 ||
-		((CCombinedFeatures*) l)->get_num_feature_obj() != ((CCombinedFeatures*) r)->get_num_feature_obj() )
-		SG_ERROR("CombinedKernel: Number of features/kernels does not match - bailing out\n")
+	if (l_combined->get_num_feature_obj() <= 0 ||
+	    l_combined->get_num_feature_obj() != r_combined->get_num_feature_obj())
+		SG_ERROR(
+		    "CombinedKernel: Number of features/kernels does not match - "
+		    "bailing out\n")
 
 	init_normalizer();
-	initialized=true;
+	initialized = true;
 	return true;
 }
 
+bool CCombinedKernel::init(CFeatures* l, CFeatures* r)
+{
+	if (enable_subkernel_weight_opt && !weight_update)
+	{
+		init_subkernel_weights();
+	}
+	/*
+	 * The two subsets, we will be passing those to init_with_extracted_subsets
```

Comment:
usually, we try to not have comments like this, a bit superflous, the code tells the story :)